src/App.tsx

```diff
@@ -1,24 +1,343 @@
 /* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import classNames from 'classnames';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import {
+  deleteTodos, getTodos, postTodos, deleteCompletedTodos,
+} from './api/todos';
 
-const USER_ID = 0;
+const USER_ID = 7094;
+
+enum Filter {
+  All = 'all',
+  Active = 'active',
+  Completed = 'completed',
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState<Filter>(Filter.All);
+  const [activeTodoId, setActiveTodoId] = useState<number>(0);
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [newTodoTitle, setNewTodoTitle] = useState<string>('');
+  const [isAddingNewTodo, setIsAddingNewTodo] = useState(false);
+
+  const showErrorNotification = (error: string) => {
+    setErrorMessage(error);
+
+    setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
+  };
+
+  const handleCloseNotification = () => {
+    setErrorMessage('');
+  };
+
+  const filteredTodos = todos.filter((todo) => {
+    switch (filter) {
+      case Filter.Active:
+        return !todo.completed;
+      case Filter.Completed:
+        return todo.completed;
+      default:
+        return true;
+    }
+  });
+
+  const handleFilterChange = (
+    option: Filter, event: React.MouseEvent<HTMLAnchorElement, MouseEvent>,
+  ) => {
+    event.preventDefault();
+    setFilter(option);
+  };
+
+  const handleTodoCheck = (idTodo: number) => {
+    setTodos((prevTodos) => prevTodos.map((todo) => {
+      if (todo.id === idTodo) {
+        return {
+          ...todo,
+          completed: !todo.completed,
+        };
+      }
+
+      return todo;
+    }));
+  };
+
+  const remainingTodos = todos.filter((todo) => !todo.completed).length;
```

Comment:
Better name it "remainingTodosCount" or "remainingTodosLength" as it stores number, not array of todos